### DISCUSSION TITLE
Mobile app?

### DISCUSSION THREAD

**Discussion by 9z0** [ransome1/sleek#215] [Ideas]
Do you have experience with mobile app development? If yes, what are your thoughts about a mobile app?

There are a couple of apps for todo.txt on Android, [Simpletask](https://github.com/mpcjanssen/simpletask-android/), [Mindstream](https://github.com/xuhcc/mindstream), [Markor](https://github.com/gsantner/markor/) (supports multiple formats), etc.

Simpletask is the best of them in my opinion but the developer had issues with Google's heavy hand and development lost a lot of stream due to lack of motivation. It is the best candidate to improve upon. Primarily, it would need to be less text-focused (no text styles, few colors, todo list looks like a wall of text especially if you have longer lines) and more of an actual GUI with buttons and separation between elements. Not difficult but would have a great effect.

**Comment by ransome1** [comment]
I don't have any knowledge on how to create a mobile app.

In general it maybe be possible to create a native frame app and somehow let sleek run inside that frame. I think this would be called a hybrid app. But as sleek is an Electron based app I'm not sure if this is possible. I read a little about it a while ago and my conclusion was to better skip it for now. Still my secret hope is that the Electron devs will provide some kind of Electron -> mobile app parser, with which output can be put into a native frame container.

For something like a sleek mobile app we would need to
- figure out if one can convert an Electron app to something usable for a mobile app
- find a Swift or Kotlin developer who could maintain the app and connect it to native functions like notifications, file system management, etc.
- work on the current state of the mobile/responsive view, which is sufficient for a desktop app, but would need improvement for a fully mobile app

**Reply by 9z0** [reply]
I see. I think that we should encourage Simpletask developer to keep improving it, seeing the success of sleek and his obvious appreciation of todo.txt !

**Comment by Saturo78** [comment]
For me simpletask works really good.
No need for another app.

**Reply by 9z0** [reply]
Unfortunately, it is too "text-like" (see my top post), and filtering/sorting could be improved.

Also, development has been very slow in the past year or two.

**Comment by lbtbs** [comment]
Simpletask has only Android version.

**Comment by new-creation7** [comment]
Check out _ntodotxt_ : https://github.com/tmaegel/ntodotxt

**Reply by lbtbs** [reply]
android only

**Comment by azzamsa** [comment]
Related #703

**Comment by adilsaribay** [comment]
Any plans for Android?

**Reply by Lezurex** [reply]
No, we don't currently have any plans on porting to a mobile app. Sleek is an Electron desktop application, as such it's built with web technologies (TS/JS, CSS, HTML). While it is generally possible to port such a project to mobile, it'd still need a massive effort regarding file operations, mobile-friendly UI etc. We currently don't have the capacity nor the experience for such an effort and the additional maintenance needed.

As for myself, I'm using the aforementioned ntodotxt as my mobile companion. It syncs nicely with my Nextcloud (or any other WebDAV service).
